File: lib/domain/usecases/branch/create_branch_usecase.dart
import '../../entities/branch.dart';
import '../../repositories/i_branch_repository.dart';

/// 创建分支用例
class CreateBranchUseCase {
  CreateBranchUseCase(this._branchRepository);

  final IBranchRepository _branchRepository;

  Future<Branch> execute({
    required String repositoryId,
    required String name,
    String? parentBranchId,
    String color = '#3B82F6',
  }) async {
    if (name.trim().isEmpty) {
      throw ArgumentError('分支名称不能为空');
    }

    final branch = Branch.create(
      repositoryId: repositoryId,
      name: name.trim(),
      parentBranchId: parentBranchId,
      color: color,
    );

    return await _branchRepository.create(branch);
  }
}

File: web/src/application/usecases/branch/create-branch-usecase.ts
import { Branch } from '../../../domain/entities/branch';
import { IBranchRepository } from '../../../domain/repositories/i-branch-repository';
import { validateNotEmpty } from '../../../core/utils/validators';

export interface CreateBranchInput {
  repositoryId: string;
  name: string;
  parentBranchId?: string | null;
  color?: string;
}

export class CreateBranchUseCase {
  constructor(private branchRepo: IBranchRepository) {}

  async execute(input: CreateBranchInput): Promise<Branch> {
    validateNotEmpty(input.repositoryId, '仓库 ID');
    validateNotEmpty(input.name, '分支名称');
    const branch = Branch.create(input.repositoryId, input.name, {
      parentBranchId: input.parentBranchId ?? null,
      color: input.color ?? '#3B82F6',
    });
    return this.branchRepo.create(branch);
  }
}
